spec/unit/pops/types/iterable_spec.rb

```diff
@@ -0,0 +1,229 @@
+require 'spec_helper'
+require 'puppet/pops'
+
+module Puppet::Pops::Types
+describe 'The iterable support' do
+
+  [5, (3..10), %w(a b c), 'hello', PIntegerType.new(1, 4), PEnumType.new(%w(yes no)) ].each do |obj|
+    it "should consider instances of #{obj.class.name} to be Iterable" do
+      expect(PIterableType::DEFAULT.instance?(obj)).to eq(true)
+    end
+
+    it "should yield an Iterable instance when Iterable.on is called with a #{obj.class.name}" do
+      expect(Iterable.on(obj)).to be_a(Iterable)
+    end
+  end
+
+  { -1 => 'a negative Integer', 0 => 'the value 0', PIntegerType.new(nil, nil) => 'an unbounded Integer type' }.each_pair do |obj, desc|
```

Comment:
This list needs a Ruby class that is Eumerable. To test that we do not accept just any Enumerable.